services/galley/src/Galley/API/Teams.hs

```diff
@@ -393,25 +392,31 @@ deleteTeam zusr zcon tid body = do
         ensureReAuthorised zusr (body ^. tdAuthPassword) (body ^. tdVerificationCode) (Just U.DeleteTeam)
 
 -- This can be called by stern
-internalDeleteBindingTeamWithOneMember ::
+internalDeleteBindingTeam ::
   Members
     '[ ErrorS 'NoBindingTeam,
+       ErrorS 'TeamNotFound,
        ErrorS 'NotAOneMemberTeam,
        ErrorS 'DeleteQueueFull,
        Queue DeleteItem,
        TeamStore
      ]
     r =>
   TeamId ->
+  Bool ->
   Sem r ()
-internalDeleteBindingTeamWithOneMember tid = do
-  team <- E.getTeam tid
-  unless ((view teamBinding . tdTeam <$> team) == Just Binding) $
-    throwS @'NoBindingTeam
-  mems <- E.getTeamMembersWithLimit tid (unsafeRange 2)
-  case mems ^. teamMembers of
-    (mem : []) -> queueTeamDeletion tid (mem ^. userId) Nothing
-    _ -> throwS @'NotAOneMemberTeam
+internalDeleteBindingTeam tid force = do
+  mbTeamData <- E.getTeam tid
+  case tdTeam <$> mbTeamData of
+    Nothing -> throwS @'TeamNotFound
+    Just team | team ^. teamBinding /= Binding -> throwS @'NoBindingTeam
+    Just team -> do
+      mems <- E.getTeamMembersWithLimit tid (unsafeRange 2)
+      case mems ^. teamMembers of
+        [mem] -> queueTeamDeletion tid (mem ^. userId) Nothing
+        -- if the team has more than one member (and deletion is forced) or no members we use the team creator's userId for deletion events
+        xs | null xs || force -> queueTeamDeletion tid (team ^. teamCreator) Nothing
+        _ -> throwS @'NotAOneMemberTeam
```

Comment:
It looks like you've rewritten some things that didn't change the behavior of the function.  This makes the commit hard to review and is a potential source of errors.  As a general rule, if you have to do rewrites, please commit them separately.

Reply:
After having read it, this looks sound.  Maybe there wasn't anything obvious to rewrite without changing semantics separately?  Except maybe the `TeamNotFound` error?